### PR TITLE
Translate Supervisor Samples in Doc to other API

### PR DESCRIPTION
**Pull request by remi-cyberbotics**
Add examples in different programming languages into supervisor documentation.

This pull-request fixes issue #1498 

https://cyberbotics.com/doc/guide/supervisor-programming?version=enhancement-translate-supervisor-doc-samples#tracking-the-position-of-robots

### CODE REVIEW

**Comment by remi-cyberbotics** [comment]
I don't know how I could test Matlab, would it be possible ? Also what do you think so far, is it looking better ?

**Comment by omichel** [comment]
That looks good to me. I would just replace "eventually" with "possibly".
About MATLAB, please try to write it blindly and I will test it (and fix it if needed) as I have a MATLAB license here.

**Comment by DavidMansolino** [comment]
> I don't know how I could test Matlab, would it be possible ? Also what do you think so far, is it looking better ?

Looks globally very good to me too.

File: docs/guide/supervisor-programming.md [hunk 1]
```diff
@@ -111,12 +229,177 @@ int main() {
...
+        robot.resetPhysics();
+      }
+    }
+  }
+}
+```
+%tab-end
+
+%tab "MATLAB"
+```matlab
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:368]:
```suggestion
```MATLAB
```

File: docs/guide/supervisor-programming.md [hunk 2]
```diff
@@ -111,12 +229,177 @@ int main() {
...
+    ret = wb_supervisor_field_get_sf_vec3f(trans_field);
+    dist = sqrt((ret[0] * ret[0]) + (ret[2] * ret[2]));
+    fprintf('a=%g b=%g dist=%g\n', a, b, dist);
+
+    % reset robot position and physics
+    INITIAL = [0, 0.5, 0];
+    wb_supervisor_field_set_sf_vec3f(trans_field, INITIAL);
+    wb_supervisor_node_reset_physics(robot_node);
+  end
+end
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:398]: Tested with MATLAB R2019a
```suggestion
TIME_STEP = 32;
robot_node = wb_supervisor_node_get_from_def('MY_ROBOT');
trans_field = wb_supervisor_node_get_field(robot_node, 'translation');
for a = 0:25
  for b = 0:33
    % evaluate robot during 60 seconds (simulation time)
    for t = 0:int64(TIME_STEP / 1000)
      % perform robot control according to a, b
      % (and possibly t) parameters.
      if wb_robot_step(TIME_STEP) == -1
        wb_robot_cleanup();
        quit(0);
      end
    end
    % compute travelled distance
    ret = wb_supervisor_field_get_sf_vec3f(trans_field);
    dist = sqrt((ret(1) * ret(1)) + (ret(3) * ret(3)));
    wb_console_print(sprintf('a=%g b=%g dist=%g\n', a, b, dist), WB_STDOUT);
    % reset robot position and physics
    INITIAL = [0, 0.5, 0];
    wb_supervisor_field_set_sf_vec3f(trans_field, INITIAL);
    wb_supervisor_node_reset_physics(robot_node);
  end
end
```

File: docs/guide/supervisor-programming.md [hunk 3]
```diff
@@ -48,6 +50,112 @@ int main() {
...
+
+  // retrieve concerned robot node from supervisor
+  Node *robot = supervisor->getFromDef("MY_ROBOT");
+
+  // retrieve concerned field from robot node
+  Field *translation = robot->getField("translation");
+
+  while (supervisor->step(timeStep) != -1) {
+    // retrieve and print values from translation field repeatedly
+    const double *values = translation->getSFVec3f();
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:76]: It's a pity you changed the variables names across languages (`trans`, `values`, `res`, etc.) this makes the correspondence difficult to understand. Also `TIME_STEP` is sometimes hard-coded to 32 and sometimes defined in a variable with different spellings... All this should be more consistent. Can you fix it?

File: docs/guide/supervisor-programming.md [hunk 4]
```diff
@@ -111,12 +229,177 @@ int main() {
...
+from controller import Supervisor
+
+supervisor = Supervisor()
+
+timestep = int(supervisor.getBasicTimeStep())
+robot_node = supervisor.getFromDef("MY_ROBOT")
+translation_field = robot_node.getField("translation")
+
+for a in range(0, 25):
+    for b in range(0, 33):
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:299]: Can't we use the same values (integers is probably simpler) in all languages. Also the comments should be the same and at the same place across different languages.

File: docs/guide/supervisor-programming.md [hunk 5]
```diff
@@ -111,12 +229,177 @@ int main() {
...
+            # (and possibly t) parameters.
+
+            # controller termination
+            if supervisor.step(timestep) == -1:
+                quit()
+
+        # compute travelled distance
+        pos = translation_field.getSFVec3f()
+        dist = sqrt(pos[0] * pos[0] + pos[2] * pos[2])
+        print("a=" + str(a) + " b=" + str(b) + " dist=" + str(dist))
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:313]: Obviously the output here is not the same as in C or C++. We should ensure it is exactly the same.

File: docs/guide/supervisor-programming.md (hunk 3 above)
Comment (remi-cyberbotics) [reply docs/guide/supervisor-programming.md:76]: Do we want to set it manually or give an example on how to retrieve it then ?

File: docs/guide/supervisor-programming.md (hunk 5 above)
Comment (remi-cyberbotics) [reply docs/guide/supervisor-programming.md:305]: It's done, there is one thing in Java though, even if the format is the same as every over languages (using the `%g` flag), the output will differ a little, leaving some trailing zeroes, I believe this is due to how Java handles floating values...

**Comment by omichel** [comment]
Accepting my suggestions would have made the review simpler for me instead of committing them separately (now I need to re-check them).

File: docs/guide/supervisor-programming.md [hunk 6]
```diff
@@ -82,41 +180,221 @@ To move the robot we need the `wb_supervisor_*` functions and hence the `supervi
...
+%tab "C++"
+```cpp
+#include <cmath>
+#include <webots/Supervisor.hpp>
+
+#define TIME_STEP 32
+
+using namespace webots;
 
+int cleanUp(Supervisor *supervisor) {
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:236]:
```suggestion
static int cleanUp(Supervisor *supervisor) {
```

File: docs/guide/supervisor-programming.md [hunk 7]
```diff
@@ -82,41 +180,221 @@ To move the robot we need the `wb_supervisor_*` functions and hence the `supervi
...
+import java.lang.Math;
+import com.cyberbotics.webots.controller.Node;
+import com.cyberbotics.webots.controller.Field;
+import com.cyberbotics.webots.controller.Supervisor;
+
+public class SupervisorController {
+
+  public static void main(String[] args) {
+
+    int TIME_STEP = 32;
```
Comment (omichel) [inline docs/guide/supervisor-programming.md:329]:
```suggestion
    final int TIME_STEP = 32;
```

**Comment by remi-cyberbotics** [comment]
> Accepting my suggestions would have made the review simpler for me instead of committing them separately (now I need to re-check them).

Sorry about that, will do next time.

Also it is nice to see those functions static, thanks.

**Pull request merged**